packages/core/src/app/shipping/MultiShippingFormV2.test.tsx

```diff
@@ -185,6 +188,51 @@ describe('MultiShippingFormV2 Component', () => {
         const allocateItemsModalHeader = within(allocateItemsModal).getByText('Destination #2');
 
         expect(allocateItemsModalHeader).toBeInTheDocument();
+
+        await waitFor(() => {
+            expect(within(allocateItemsModal).queryByText('Product 1')).not.toBeInTheDocument();
+        });
+        expect(within(allocateItemsModal).getByText('Product 2')).toBeInTheDocument();
+
+        const physicalItemQuantityInput = within(allocateItemsModal).getByLabelText('Quantity of Product 2');
+
+        expect(physicalItemQuantityInput).toBeInTheDocument();
+        expect(physicalItemQuantityInput).toHaveValue(0);
+        expect(within(allocateItemsModal).getByRole('button', { name: 'Allocate' })).toBeDisabled();
+
+        await userEvent.type(physicalItemQuantityInput, '5');
+
+        expect(physicalItemQuantityInput).toHaveValue(5);
+        expect(within(allocateItemsModal).getByRole('button', { name: 'Allocate' })).toBeEnabled();
+
+        await userEvent.click(within(allocateItemsModal).getByRole('button', { name: 'Allocate' }));
+        expect(within(allocateItemsModal).getByText('Quantity cannot exceed "Left to allocate" amount.')).toBeInTheDocument();
+
+        await userEvent.clear(physicalItemQuantityInput);
+        await userEvent.type(physicalItemQuantityInput, '1');
+        expect(physicalItemQuantityInput).toHaveValue(1);
+
+        const customItemQuantityInput = within(allocateItemsModal).getByLabelText('Quantity of Custom item');
+
+        expect(customItemQuantityInput).toBeInTheDocument();
+        expect(customItemQuantityInput).toHaveValue(0);
+
+        await userEvent.type(customItemQuantityInput, '5');
+
+        expect(customItemQuantityInput).toHaveValue(5);
+
+        await userEvent.click(within(allocateItemsModal).getByRole('button', { name: 'Allocate' }));
+        expect(within(allocateItemsModal).getByText('All quantities of this custom product must be allocated to the same destination, as it was created specifically for this draft order.')).toBeInTheDocument();
```

Comment:
For a message that is eactly the same as the one in json file, we can get the string by read the value like this `localeContext.language.translate('payment.instrument_manage_delete_server_error')`.

The rest looks good to me.